app/controllers/subscriptions_controller.rb

```diff
@@ -1,5 +1,18 @@
 class SubscriptionsController < ApplicationController
-  before_action :find_rubygem
+  before_action :redirect_to_signin, only: :index, unless: :signed_in?
+  before_action :redirect_to_new_mfa, only: :index, if: :mfa_required_not_yet_enabled?
+  before_action :redirect_to_settings_strong_mfa_required, only: :index, if: :mfa_required_weak_level_enabled?
```

Comment:
Forcing strong MFA isn't necessary here (or even forcing MFA in general), subscriptions are not super critical and aren't proportional to the actions being taken here.

Reply:
I copied this from the dashboard controller since that's where this used to be shown, but I agree. We can remove it.

Reply:
I looked into this further and we have a pattern of redirecting to setting up strong MFA on every page that I could find that requires you to be signed in. It seems to me we just demand that users do nothing else until their MFA is updated when it is required.

Reply:
Given that, I think we should refactor our login check to basically run through all the requirements in a single before_action so that this duplicate code can be cleaned up. Even upon fresh sign in we do the same check, so the pattern is really: "Are you not signed in? => Sign in page; Is your MFA inadequate? => MFA page. Otherwise continue..."